src/external/curl/CMakeLists.txt

```diff
@@ -72,8 +72,8 @@ set(OS "\"${CMAKE_SYSTEM_NAME}\"")
 include_directories(${PROJECT_BINARY_DIR}/include/curl)
```

Comment:
I propose to rename this file and create new one. (as I did for zlib and chipmunk) Currently most of libraries even if they have cmake is definitely not "modern cmake" and have lots of garbage...
My plan is to create PRs to libraries author (zlib, chipmunk etc) with "modern cmake". And than we will use CMakeLists.txt without any modifications.

Reply:
that will be perfect. I have a doubt when you use "modern cmake" on zlib, you lost some macro defines, for example:

```
	if(ZLIB_ASMS)	
		add_definitions(-DASMV -DASMINF)	
	endif()
```
it was in `CMakeLists.txt.org`, but not new added `CMakeLists.txt`. I think it maybe cause hidden issues ? since we didn't know what's the effect of this macro.

Reply:
Please search zlib code for "bugs". Generally it is not recommended to use assembler code for zlib.

But I agree we have to be careful to don't miss some important settings ....